### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by rosenhouse**
We were seeing some flakes when running the test suite on our AWS box with Go 1.5

  These changes seem to help.

cc: @zachgersh

### CODE REVIEW

**Comment by glestaris** [comment]
Thanks for the fixes! This time you were faster than us (https://www.pivotaltracker.com/story/show/110174588) :)

Can we have the `DestroyAndStop` call in a suite `AfterEach`? Presumably this means that the client variable has to be global, but this is fine. I am concerned that we might add new files in GQT and forget to call `DestroyAndStop`.

**Comment by rosenhouse** [comment]
Hey @glestaris,

I've added a commit that consolidates all the `DestroyAndStop` calls into a single suite-level `AfterEach`.  

However, I have some reservations about that.  It visually separates the creation from the destruction, with the former appearing as `createGarden()` in each test file, and the latter living as a single `DestroyAndStop` in the common `gqt_suite_test.go` file.  So a contributor must look in two different places to understand the full lifecycle of the client.  In my opinion, that makes the tests harder to read and reason about.

You might suggest that the `startGarden()` calls also live in a suite-level `BeforeEach`.  Then the full lifecycle would be defined in the `gqt_suite_test.go` file.  But that does not work well with the [`garden_api_test.go` file](https://github.com/cloudfoundry-incubator/guardian/blob/master/gqt/garden_api_test.go#L15-L46), in which the arguments to guardian are customized at the `BeforeEach` level, and the `startGarden(args...)` is called in a `JustBeforeEach`.

I can see two ways to work around that issue:
  - in `garden_api_test`'s `JustBeforeEach`, destroy the suite-created `client` and recreate it
  - change the global `startGarden(args...)` to live in a suite-level `JustBeforeEach` instead of `BeforeEach`

Neither one is awesome.  The latter requires a cascade of changes to other tests, as `BeforeEach`s in various tests which assume that the global `client` has been initialized now much be changed into `JustBeforeEach`s.

In summary, I'm somewhat partial to the code as it was before (ee89dd3) but I've made the changes that you suggested, and am ok with a merge either way.

**Comment by glestaris** [comment]
Hello @rosenhouse,

thanks for sharing your thoughs. I think your view is valuable because my suggestion was biased from the way we were doing things in the garden-linux equivalent of GQT. I agree that the separation of client creation and destruction is not very clean and it will confuse users. Additionally any solution that includes use of `JustBeforeEach` in the suite decreases the expressiveness you can achieve in the contexts and describes.

I am convinced that we should go with your original approach. We will make sure, in the Garden team, that we don't forget to add `DestroyAndStop` in new tests.

**Comment by julz** [comment]
I'm worried that we didn't spot the absence of these AfterEachs for multiple weeks while they slowly made our tests slower. How will we prevent this happening again? Could we maybe do a sanity check in an AfterSuite to check that the client is either nil or stopped? (Having done the sanity check we might  wonder why the sanity check doesn't just do a just-in-case destroy - which should be idempotent - and then why we're bothering to do lots of destroys when that one would do, but that's another matter).

**Comment by glestaris** [comment]
@julz this is why I suggested having it in the suite `AfterEach` in the beginning. However, if you think about it, the only reason we failed to clean up is because we are used to the garden-linux integration tests' way of cleaning up the client. I am sure that if we communicate this properly we will be cleaning it up as part of new test files, the same way we delete temporary files in `AfterEach`. It is a good practice.

@rosenhouse's argument is, I think, very interesting, especially since he represents contributors outside of the Gaden team who are unaware of this convention.

**Comment by julz** [comment]
Ok, how about we try it this way (with lots of AfterEaches) and if we start leaking again we can revisit whether a suite-level aftereach is a good idea.

**Comment by glestaris** [comment]
Sure, this is uncontroversial. If we observe new leaks we should reconsider.

**Comment by rosenhouse** [comment]
Ok, removed the "consolidate" commit & rebased.  should be good to go.

**Comment by svett** [comment]
@rosenhouse / @zachgersh Thanks for this contribution.

- @svett & @glestaris
